.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,162 @@
+env:
+  ARCH: amd64
+  TAG: v2.8-${{ github.sha }}
+  GIT_TAG: v2.8-${{ github.sha }}
+  HEAD_TAG: v2.8-${{ github.sha }}
+  COMMIT: ${{ github.sha }}
+  REPOSITORY_OWNER: ${{ github.repository_owner }}
+  IMAGE: ${{ github.repository_owner }}/rancher
+  IMAGE_AGENT: ${{ github.repository_owner }}/rancher-agent
+  IMAGE_INSTALLER: ${{ github.repository_owner }}/system-agent-installer-rancher
+  CATTLE_KDM_BRANCH: dev-v2.8
+  HELM_UNITTEST_VERSION: 0.3.2
+  HELM_URL_V3: https://get.helm.sh/helm-v3.13.3-linux-amd64.tar.gz
+jobs:
+  build-chart:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        run: |
+          sudo snap install yq
+          curl ${{ env.HELM_URL_V3 }} | tar xvzf - --strip-components=1 -C /tmp/ && \
+          sudo mv /tmp/helm /usr/bin/helm_v3 && \
+          sudo chmod +x /usr/bin/helm_v3
+          helm_v3 plugin install https://github.com/helm-unittest/helm-unittest.git --version ${{ env.HELM_UNITTEST_VERSION }}; \
+      - name: Build
+        run: ./scripts/chart/build chart
+      - name: Validate
+        run: ./scripts/chart/validate
+      - name: Test
+        run: ./scripts/chart/test
+      - name: Package
+        run: ./scripts/chart/package
+  build-server:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [linux]
+        arch: [amd64, arm64]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup Environment Variables
+        run: |
+          echo "ARCH=${{ matrix.arch }}" >> "$GITHUB_ENV"
+          echo "RKE_VERSION=$(grep -m1 'github.com/rancher/rke' go.mod | awk '{print $2}')" >> "$GITHUB_ENV"
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.IMAGE }}
+          flavor: |
+            latest=false
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Create k3s images file
+        run: touch k3s-airgap-images.tar
+      - name: Download data.json
+        run: curl -sLf https://releases.rancher.com/kontainer-driver-metadata/${{ env.CATTLE_KDM_BRANCH }}/data.json > ./data.json
+      - name: Build and export server
+        id: build
+        uses: docker/build-push-action@v5
+        with:
+          push: false
+          build-args: |
+            "VERSION=${{ env.COMMIT }}"
+            "COMMIT=${{ env.COMMIT }}"
+            "RKE_VERSION=${{ env.RKE_VERSION }}"
+            "ARCH=${{ matrix.arch }}"
+          tags: ${{ env.IMAGE }}:${{ env.TAG }}-${{ matrix.arch }}
+          context: .
+          platforms: "${{ matrix.os }}/${{ matrix.arch }}"
+          file: ./package/Dockerfile
+          labels: "${{ steps.meta.outputs.labels }}"
+          outputs: type=docker,dest=/tmp/rancher-${{ matrix.os }}-${{ matrix.arch }}.tar
+      - name: Upload image
```

Comment:
Are we pushing images to dockerhub every time a PR is opened? Right now we only push when a merge happens, right?

Reply:
Never mind, realized this doesn't push anything to dockerhub and is being used to build locally